File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {

        int n = nums.size();

        int  maxJump = 0;
        int j = 0;

        for(int i = 0; i < nums.size(); i++){

            if(i > maxJump){
                return false;
            }

            j = i + nums[i];

            maxJump = max(maxJump, j);
        }

        return maxJump >= n-1;
        
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int jumpMax = 0;
        int j = 0;
        for(int i = 0;i<nums.length;i++){
            if(i > jumpMax){
                return false;
            }
            j = i+nums[i];
            jumpMax = Math.max(jumpMax,j);
        }
        return jumpMax >= n-1;
        
    }
}